2.0/flask_app/services/vector_search.py

```
@@ -0,0 +1,91 @@
+                        s.text, 
+                        f.date, 
+                        f.type,
+                        f.company_name, 
+                        1 - (e.embedding <=> CAST(:embedding AS vector)) AS cosine_similarity
+                    FROM {embed_table_name} e
+                    JOIN {table_name} s ON e.section_id = s.id
+                    JOIN {settings.FILING_INFO_TABLE} f ON s.accession_number = f.accession_number
+                    WHERE 1 - (e.embedding <=> CAST(:embedding AS vector)) > :threshold
+                    ORDER BY cosine_similarity DESC
+                    LIMIT :top_k
+                """)
```

Ruff (0.13.3)

Possible SQL injection vector through string-based query construction

(S608)